docs/development.md

```diff
@@ -35,3 +35,15 @@ Since this is very unlikely and most times an abs path is not strictly necessary
 "Should" functions can be used anywhere,
 but care should be taken not to overuse them when a fallback value is not sufficient.
 "Must" variants should be favored in testing code, since they will more obviously fail tests.
+
+## Do/Don't for cross-platform shell scripting
```

Comment:
There is some general structural work for this section that I am going to put here in a single comment.
Overall, we want this section to be more friendly/accessible.

Start with changing the title to something like "Cross-Platform Scripting".

Then, ass in an introduction paragraph that says something about why this section needs to exist.
Like "we are writing bash scripts with the intention of being run in a POSIX environment, but there are still some differences you need to be aware of ...". 

Then, break it up by tools.
For example, `-P`/`-E` are specifically for grep.